test/python/test_composite.py

```diff
@@ -1081,159 +1088,157 @@ def test_input_data_user_modifications(self):
-        self.assertArrayNear(gas.concentrations, gas2.concentrations)
-        self.assertArrayNear(gas.partial_molar_enthalpies,
-                             gas2.partial_molar_enthalpies)
-        self.assertArrayNear(gas.forward_rate_constants,
-                             gas2.forward_rate_constants)
-        self.assertArrayNear(gas.mix_diff_coeffs, gas2.mix_diff_coeffs)
-
-    def test_yaml_outunits2(self):
+        assertArrayNear(gas.concentrations, gas2.concentrations)
+        assertArrayNear(gas.partial_molar_enthalpies,
+                        gas2.partial_molar_enthalpies)
+        assertArrayNear(gas.forward_rate_constants,
+                        gas2.forward_rate_constants)
+        assertArrayNear(gas.mix_diff_coeffs, gas2.mix_diff_coeffs)
+
+    def test_yaml_outunits2(self, load_yaml):
         gas = ct.Solution('h2o2.yaml')
         gas.TPX = 500, ct.one_atm, 'H2: 1.0, O2: 1.0'
         gas.equilibrate('HP')
         gas.TP = 1500, ct.one_atm
         units = {'length': 'cm', 'quantity': 'mol', 'energy': 'cal'}
         system = ct.UnitSystem(units)
         gas.write_yaml(self.test_work_path / "h2o2-generated.yaml", units=system)
-        generated = utilities.load_yaml(self.test_work_path / "h2o2-generated.yaml")
-        original = utilities.load_yaml(self.cantera_data_path / "h2o2.yaml")
+        generated = load_yaml(self.test_work_path / "h2o2-generated.yaml")
+        original = load_yaml(self.cantera_data_path / "h2o2.yaml")
 
         for r1, r2 in zip(original['reactions'], generated['reactions']):
             if 'rate-constant' in r1:
-                self.assertNear(r1['rate-constant']['A'], r2['rate-constant']['A'])
-                self.assertNear(r1['rate-constant']['Ea'], r2['rate-constant']['Ea'])
+                assertNear(r1['rate-constant']['A'], r2['rate-constant']['A'])
+                assertNear(r1['rate-constant']['Ea'], r2['rate-constant']['Ea'])
 
         gas2 = ct.Solution(self.test_work_path / "h2o2-generated.yaml")
-        self.assertArrayNear(gas.concentrations, gas2.concentrations)
-        self.assertArrayNear(gas.partial_molar_enthalpies,
-                             gas2.partial_molar_enthalpies)
-        self.assertArrayNear(gas.forward_rate_constants,
-                             gas2.forward_rate_constants)
-        self.assertArrayNear(gas.mix_diff_coeffs, gas2.mix_diff_coeffs)
-
-    def check_ptcombust(self, gas, surf):
-        generated = utilities.load_yaml(self.test_work_path / "ptcombust-generated.yaml")
+        assertArrayNear(gas.concentrations, gas2.concentrations)
+        assertArrayNear(gas.partial_molar_enthalpies,
+                        gas2.partial_molar_enthalpies)
+        assertArrayNear(gas.forward_rate_constants,
+                        gas2.forward_rate_constants)
+        assertArrayNear(gas.mix_diff_coeffs, gas2.mix_diff_coeffs)
+
+    @pytest.mark.usefixtures('load_yaml')
+    def check_ptcombust(self, gas, surf, load_yaml):
```

Comment:
Why is `@pytest.mark.usefixtures` needed here, if the calls to `check_ptcombust` already pass in this fixture as received from the `test_*` methods? Or is this a way to make specifying the `load_yaml` argument explicitly unnecessary?

Reply:
This was a remnant of me fighting with pytest fixtures to get the yaml fixture to work. I removed it.